camp/jbolotski/beam_break_motor/src/main/java/frc/robot/sim/PhysicsSim.java

```diff
@@ -0,0 +1,106 @@
+package frc.robot.sim;
+
+import java.util.*;
+import com.ctre.phoenix.motorcontrol.can.*;
+
+/**
+ * Manages physics simulation for CTRE products.
+ * from https://github.com/CrossTheRoadElec/Phoenix-Examples-Languages/blob/master/Java%20General/MotionMagic/src/main/java/frc/robot/sim/PhysicsSim.java
+ */
+public class PhysicsSim {
```

Comment:
Cool stuff ... I will read this tomorrow ... getting tired need to go to bed.

Reply:
What I've found since posting the PR is that the simulation is able to run without this custom PhysicsSim plugin.  But, the motor that's simulated is very naive e.g. it immediately goes to the exact output setting that you command.  With this plugin, there is some spin up time for the motor and there is some noise on the output value.  For debugging just to get code working, I don't think we need this realism.  We could elect to use this plugin only when we want/need that realism.   
I also can't figure out how to use the debugger with the simulator.  It seems like the whole point of the simulator is to be able to debug.  So, I feel like I'm missing something really obvious.

Reply:
Just tried it out for first time.  Likely need to keep playing with it more.